hooks/pre/10-reset_data.rb

```diff
@@ -2,22 +2,22 @@
 require 'English'
 
 def reset
-  if foreman_installed?
+  if foreman_maintain_present?
 
     stop_services
-    start_services(['postgresql'])
-    reset_database
-    reset_candlepin
-    reset_pulp
+    start_services(['postgresql']) if foreman_server? && local_postgresql?
```

Comment:
For now, I could see wrapping `stop_services` and `start_services` in the if foreman-maintain present check. 

Will clearing the Foreman database cover any Dynflow related data? @adamruzicka

Reply:
I can answer that @ehelms, the answer is yes it will.

Dynflow data lives in these tables: dynflow_envelopes,dynflow_delayed_plans,dynflow_steps,dynflow_actions,dynflow_execution_plans,foreman_tasks_locks,foreman_tasks_tasks

All of those reside in the main foreman DB

Reply:
With sidekiq in place, it would be good to restart redis so it gets cleared as well

Reply:
Thanks @adamruzicka . With full service restarts in place, we should be good then.